dbunit/src/test/java/com/baeldung/dbunit/PrepAndExpectedDbUnitTest.java

```diff
@@ -0,0 +1,103 @@
+package com.baeldung.dbunit;
+
+import org.dbunit.*;
+import org.dbunit.dataset.IDataSet;
+import org.dbunit.dataset.xml.FlatXmlDataSetBuilder;
+import org.dbunit.operation.DatabaseOperation;
+import org.dbunit.util.fileloader.DataFileLoader;
+import org.dbunit.util.fileloader.FlatXmlDataFileLoader;
+import org.junit.Test;
+
+import java.io.InputStream;
+import java.sql.Connection;
+import java.sql.ResultSet;
+
+public class PrepAndExpectedDbUnitTest extends DefaultPrepAndExpectedTestCase {
+    private static final String JDBC_DRIVER = org.h2.Driver.class.getName();
+    private static final String JDBC_URL = "jdbc:h2:mem:default;DB_CLOSE_DELAY=-1;init=runscript from 'classpath:schema.sql'";
+    private static final String USER = "sa";
+    private static final String PASSWORD = "";
+
+    @Override
+    public void setUp() throws Exception {
+        setDatabaseTester(initDatabaseTester());
+        setDataFileLoader(initDataFileLoader());
+        super.setUp();
+    }
+
+    private IDatabaseTester initDatabaseTester() throws Exception {
+        final JdbcDatabaseTester tester = new JdbcDatabaseTester(JDBC_DRIVER, JDBC_URL, USER, PASSWORD);
+        tester.setDataSet(initDataSet());
+        tester.setSetUpOperation(DatabaseOperation.REFRESH);
+        return tester;
+    }
+
+    private IDataSet initDataSet() throws Exception {
+        final InputStream is = getClass().getClassLoader().getResourceAsStream("data.xml");
+        return new FlatXmlDataSetBuilder().build(is);
+    }
+
+    private DataFileLoader initDataFileLoader() {
+        return new FlatXmlDataFileLoader();
+    }
+
+    @Test
+    public void testSelect() throws Exception {
+        // Arrange
+        final Connection connection = getConnection().getConnection();
+        final VerifyTableDefinition[] verifyTables = {new VerifyTableDefinition("USERS", new String[]{})};
+        final String[] prepDataFiles = {"/users.xml"};
+        final String[] expectedDataFiles = {"/users.xml"};
+        final PrepAndExpectedTestCaseSteps testSteps = () -> {
+            // invoke the method being tested here; for the sake of simplicity we use JDBC API directly in this example
+            final ResultSet rs = connection.createStatement().executeQuery("select * from USERS where id = 1");
+
+            // either place assertions here
+            //assertTrue(rs.next());
+            //assertEquals("Xavier", rs.getString("last_name"));
+
+            return rs;
+        };
+
+        // Act
+        final ResultSet rs = (ResultSet) super.runTest(verifyTables, prepDataFiles, expectedDataFiles, testSteps);
```

Comment:
I'm a bit confused here about the mix of assertions and `expectedDataFiles` - what does the expectedDataFiles behaviour of the `runTest` method NOT assert that you might do in some additional assertions on the results set?